### PR TITLE
Add descriptions to privilege table

### PR DESCRIPTION
**Pull request by ianjevans**
- [x] Content review
- [x] Backport to previous versions

Fixes DOC-7084.

### CODE REVIEW

File: src/current/_includes/v23.1/sql/privileges.md [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-`VIEWACTIVITY` | System
-`VIEWACTIVITYREDACTED` | System
-`VIEWCLUSTERMETADATA` | System
-`VIEWCLUSTERSETTING` | System
-`VIEWDEBUG` | System
-`VIEWJOB` | System
-`ZONECONFIG` | Database, Table, Sequence
+Privilege | Levels | Description
+----------|--------|------------
+`ALL` | Database, Schema, Table, Sequence, Type | Grants all privileges at the database, schema, table, sequence, or type level.
```
Comment (rafiss) [inline src/current/_includes/v23.1/sql/privileges.md:3]: nit: ALL also works at the "system" level

File: src/current/_includes/v23.1/sql/privileges.md [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
+`RESTORE` | System, Database | Grants the ability to restore [backups]({% link {{ page.version.version }}/backup-and-restore-overview.md %}) at the system or database level.
+`SELECT` | Table, Sequence | Grants the ability to run [selection queries]({% link {{ page.version.version }}/query-data.md %}) at the table or sequence level.
+`UPDATE` | Table, Sequence | Grants the ability to run [update statements]({% link {{ page.version.version }}/update-data.md %}) at the table or sequence level.
+`USAGE`  | Function, Schema, Sequence, Type | Grants the ability to use [functions]({% link {{ page.version.version }}/functions-and-operators.md %}), [schemas]({% link {{ page.version.version }}/schema-design-overview.md %}), [sequences]({% link {{ page.version.version }}/create-sequence.md %}), or [user-defined types]({% link {{ page.version.version }}/create-type.md %}).
+`VIEWACTIVITY` | System | Grants the ability to view other user's activity statistics of a cluster.
+`VIEWACTIVITYREDACTED` | System | Grants the ability to view other user's activity statistics, but prevents the role from accessing the statement diagnostics bundle.
```
Comment (rafiss) [inline src/current/_includes/v23.1/sql/privileges.md:23]: perhaps also mentions that certain columns in introspection queries will be redacted. specifically, those that may contain data from the cluster (e.g. `crdb_internal.node_statement_statistics(last_error)`)

**Review by rafiss — APPROVED** [review]
lgtm, just small nits

**Review by rmloveland — APPROVED** [review]
LGTM!

File: src/current/_includes/v22.2/sql/privileges.md [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
+`UPDATE` | Table, Sequence | Grants the ability to run [update statements]({% link {{ page.version.version }}/update-data.md %}) at the table or sequence level.
+`USAGE`  | Function, Schema, Sequence, Type | Grants the ability to use [functions]({% link {{ page.version.version }}/functions-and-operators.md %}), [schemas]({% link {{ page.version.version }}/schema-design-overview.md %}), [sequences]({% link {{ page.version.version }}/create-sequence.md %}), or [user-defined types]({% link {{ page.version.version }}/create-type.md %}).
+`VIEWACTIVITY` | System | Grants the ability to view other user's activity statistics of a cluster.
+`VIEWACTIVITYREDACTED` | System | Grants the ability to view other user's activity statistics, but prevents the role from accessing the statement diagnostics bundle and viewing some introspection queries that contain data about the cluster.
+`VIEWCLUSTERMETADATA` | System | Grants the ability to view range information, data distribution, store information, and Raft information.
+`VIEWCLUSTERSETTING` | System | Grants the ability to view [cluster settings]({% link {{ page.version.version }}/cluster-settings.md %}).
```
Comment (kevin-v-ngo) [inline src/current/_includes/v22.2/sql/privileges.md:25]: Can we clarify whether a user can view the cluster setting **and their values**?

File: src/current/_includes/v22.2/sql/privileges.md [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
+`SELECT` | Table, Sequence | Grants the ability to run [selection queries]({% link {{ page.version.version }}/query-data.md %}) at the table or sequence level.
+`UPDATE` | Table, Sequence | Grants the ability to run [update statements]({% link {{ page.version.version }}/update-data.md %}) at the table or sequence level.
+`USAGE`  | Function, Schema, Sequence, Type | Grants the ability to use [functions]({% link {{ page.version.version }}/functions-and-operators.md %}), [schemas]({% link {{ page.version.version }}/schema-design-overview.md %}), [sequences]({% link {{ page.version.version }}/create-sequence.md %}), or [user-defined types]({% link {{ page.version.version }}/create-type.md %}).
+`VIEWACTIVITY` | System | Grants the ability to view other user's activity statistics of a cluster.
+`VIEWACTIVITYREDACTED` | System | Grants the ability to view other user's activity statistics, but prevents the role from accessing the statement diagnostics bundle and viewing some introspection queries that contain data about the cluster.
```
Comment (kevin-v-ngo) [inline src/current/_includes/v22.2/sql/privileges.md:23]: > prevents the role from accessing the statement diagnostics bundle
If i recall, we only did work to hide access to statement bundle collection **in the console**. Please correct me here @maryliag 

> viewing some introspection queries that contain data about the cluster.
What does this mean?

VIEWACTIVITYREDACTED redacts information on the following console pages:

- Statements on the Active executions tab
- Statements on Sessions
- Statements on Workload Insights tab

**Review by kevin-v-ngo — COMMENTED** [review]
Thanks Ian! Added a few comments. I'll also share with Cluster Obs team to help review

File: src/current/_includes/v22.2/sql/privileges.md (hunk 4 above)
Comment (maryliag) [reply src/current/_includes/v22.2/sql/privileges.md:23]: That is correct, we only hide the bundle information on the Console, they can still see it using cli

Comment (ianjevans) [reply src/current/_includes/v22.2/sql/privileges.md:23]: >> viewing some introspection queries that contain data about the cluster.
>What does this mean?

@rafiss left a comment to add this:
https://github.com/cockroachdb/docs/pull/17678#discussion_r1297348442

**Review by data-matt — CHANGES_REQUESTED** [review]
Folks,

**First point**, I don't believe this one is correct:
` CONNECT | Database | Grants the ability to connect to the database.`

Unless something has changed, I believe it just allows access to table metadata, you can connect to the database with/without CONNECT. 

We differ from pg here. 

**Second point**

I think ALL is a bit confusing. Folks could mis-interpret this. 

ALL | System, Database, Schema, Table, Sequence, Type | Grants all privileges at the system, database, schema, table, sequence, or type level.
-- | -- | --

Can we put something more specific like,  "For the object the privilege is applied to, the ALL grant provides all privileges at the system, database, schema, table, sequence, or type level"

Needs a bit of word smithing, but I think its good to point out that essentially it only applies to the object ALL is granted on.

Thanks

File: src/current/_includes/v23.1/sql/privileges.md [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
-`VIEWJOB` | System
-`ZONECONFIG` | Database, Table, Sequence
+Privilege | Levels | Description
+----------|--------|------------
+`ALL` | System, Database, Schema, Table, Sequence, Type | Grants all privileges at the system, database, schema, table, sequence, or type level.
+`BACKUP` | System, Database, Table | Grants the ability to create [backups]({% link {{ page.version.version }}/backup-and-restore-overview.md %}) at the system, database, or table level.
+`CANCELQUERY` | System | Grants the ability to cancel queries.
+`CHANGEFEED` | Table | Grants the ability to create [changefeeds]({% link {{ page.version.version }}/change-data-capture-overview.md %}) on a table.
+`CONNECT` | Database | Grants the ability to connect to the database.
+`CREATE` | Database, Schema, Table, Sequence | Grants the ability to create objects at the database, schema, table, or sequence level.
```
Comment (dikshant) [inline src/current/_includes/v23.1/sql/privileges.md:8]: Perhaps we should add a bit here that says having CREATE also grants the ability to change multi region zone configs. Regardless of whether ZONECONFIG is granted or not.

Comment (dikshant) [reply src/current/_includes/v23.1/sql/privileges.md:8]: Thoughts @rafiss ?

Comment (rafiss) [reply src/current/_includes/v23.1/sql/privileges.md:8]: that sounds good. you mean CREATE at the database level right?

Comment (data-matt) [reply src/current/_includes/v23.1/sql/privileges.md:8]: +1, I think its good to point out that CREATE allows access to ZONECONFIG.

Comment (ianjevans) [reply src/current/_includes/v23.1/sql/privileges.md]: Added. And @data-matt I'll need your approval to get this merged, as you requested changes.

**Review by data-matt — APPROVED** [review]

**Pull request merged**